src/main/java/com/bookstore/service/cart/ShoppingCartServiceImpl.java

```diff
@@ -79,16 +77,6 @@ public ShoppingCartResponseDto update(Long id, CartItemRequestDtoWithoutBookId r
 
     @Override
     public void deleteItem(Long id) {
-        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
-        ShoppingCart cart = shoppingCartRepository.findById(user.getId())
-                .orElseThrow(() -> new EntityNotFoundException("Can't find shopping cart by id: "
-                        + user.getId()));
-        CartItem cartItem = cart.getCartItems().stream()
-                .filter(c -> Objects.equals(c.getId(), id))
-                .findFirst()
-                .orElseThrow(() -> new EntityNotFoundException(
-                        "Can't find cart item by id: " + id));
-        cartItem.setDeleted(true);
-        shoppingCartRepository.save(cart);
+        cartItemRepository.deleteById(id);
```

Comment:
you need to validate that item belongs to the user by calling `existsByIdAndShoppingCartId`